### ISSUE TITLE
Add support for specifying CPU counts and Total Memory

### ISSUE THREAD

**Issue by ja6a** [skytap/terraform-provider-skytap#3]
As a user I want to specify the number of vCPUs the VM has use of. Additionally, I want to set the total memory a VM may allocate. I want to set these in both create and update on the VM resource.

Scenario:

```
resource "skytap_vm" "vm" {
  "environment_id" = "${skytap_environment.my_new_environment.id}"
  "template_id"    = "1"
  "vm_id"          = "2"
  "name"           = "my_vm"
  
  "cpus" = 1 # valid 1 - 12 optional default 1 force new
  "ram" = 256 # valid 256 and 131,072 (MB) optional default 256 force new
}
```
These properties will be added at the top level of the vm. A hardware section will not be used. 
If an update is made possible (force new removed) then the VM should be stopped prior to updating either of these properties. 

**CPU** 
The template will be read during the creation process. The `max_cpus` property will be read and if the `cpus` property is over this value an error will be raised.

**RAM** 
The template will be read during the creation process. The maximum depends on the read only `max_ram` property. The minimum depends on the read only `min_ram` property. If the `ram` property is outside these bounds then an error will be raised. MB is assumed.

**Comment by ja6a** [comment]
PR #6

**Comment by ja6a** [comment]
**Note Regarding Force New**
Changing the RAM and CPU settings requires the VM to be shutdown. This is because the OS needs to boot up in order to process these settings. In terraform the standard approach of cloud providers when changing these resources is to recreate the VM resource, e.g., AWS, GCP, and Azure. Since this is an official provider we would like to stay with the best practise for these OS level changes. To clarify: the approach of changing the provider to stop and start the VM when these settings change - silently from the users point of view - will not be supported. The detailed reason is as follows: terraform has no knowledge of the application layer. It will not handle gracefully any application level considerations to perform a clean shutdown and restart.

**Comment by ja6a** [comment]
Resolved in PRs #6 and #14
